File: DataStreams.cpp
class SummaryRanges {
public:
          set<int>values;

          SummaryRanges(){};

          void addValue(int value){
               values.insert(value);
          }

          vector<vector<int>>getIntervals(){
              if(values.empty())
                  return {};

              vector<vector<int>>intervals;

              int left = -1, right = -1;

              for(auto it: values){
                   if(left < 0){
                      left = right = it;
                   }else if(it == right + 1){
                      right = it;
                   }else{
                      intervals.push_back({left,right});
                      left = right = it;
                   }
              }

              intervals.push_back({left,right});

              return intervals;
          }

};

File: DataStreams.py
class SummaryRanges:
    def __init__(self):
        self.values = set()

    def addValue(self, value):
        self.values.add(value)

    def getIntervals(self):
        if not self.values:
            return []

        intervals = []

        left, right = -1, -1

        for it in sorted(self.values):
            if left < 0:
                left = right = it
            elif it == right + 1:
                right = it
            else:
                intervals.append([left, right])
                left = right = it

        intervals.append([left, right])

        return intervals
